packages/components/src/components/as-category-widget/as-category-widget.tsx

```diff
@@ -0,0 +1,195 @@
+    };
+
+    return (
+      <li class={cssClasses} onClick={() => this._toggleCategory(category)}>
+        <p class='as-category-widget__category-title as-body' data-value={readableNumber(category.value)}>
+          {category.name}
+        </p>
+
+        <div class='as-category-widget__bar'>
+          <div class='as-category-widget__value' style={progressStyles}></div>
+        </div>
+      </li>
+    );
+  }
+
+  private _renderOtherCategory(category: Category, options: CategoryOptions) {
+    const categoryData = category || {
+      name: 'Other',
+      value: this._getCategoriesTotalValue(
+        this.categories.slice(this.numberOfVisibleCategories, this.categories.length)
+      )
+    };
+
+    return this._renderCategory(categoryData, { maximumValue: options.maximumValue, isOther: true });
+  }
+
+  private _renderFooter() {
+    const selectedCount = this.selectedCategories.length ? this.selectedCategories.length : 'All';
+
+    return (
+      <footer class='as-category-widget__footer'>
+        <div class='as-category-widget__count as-body'>{selectedCount} selected</div>
+        { this.showClearButton ? <button class='' onClick={() => this.clearSelection()}>Clear selection</button> : '' }
+      </footer>
+    );
+  }
+
+  private _isSelected(categoryName: string) {
+    return this.selectedCategories.indexOf(categoryName) > -1;
+  }
+
+  private _toggleCategory(category: Category) {
+    const newSelectedCategories = this._isSelected(category.name)
+      ? this.selectedCategories.filter((currentCategory) => currentCategory !== category.name)
+      : [...this.selectedCategories, category.name];
+
+    this.selectedCategories = newSelectedCategories;
+    this._onCategoriesChanged();
+  }
+
+  private _onCategoriesChanged() {
+    this.categoriesSelected.emit(this.selectedCategories);
+  }
+
+  private _getVisibleCategoriesMaximumValue() {
+    const categories = this.categories.slice(0, this.numberOfVisibleCategories);
+
+    return categories.reduce(
+      (maximum, currentCategory: Category) => currentCategory.value > maximum ? currentCategory.value : maximum, 0
+    );
+  }
+
+  private _getCategoriesTotalValue(categories: object[]) {
+    return categories.reduce(
+      (maximum, currentCategory: Category) => currentCategory.value + maximum, 0
+    );
+  }
+
+  private _getBarColor(color: string, options: { isSelected?: boolean } = {}) {
+    if (options.isSelected) {
+      return shadeOrBlend(-0.16, color);
+    }
+
+    return color;
+  }
+
+  private parseCategories() {
+    const newCategories = [...this.categories];
+    const otherCategoryIndex = this.categories.findIndex((category: Category) => category.name === OTHER_CATEGORY_NAME);
+
+    if (otherCategoryIndex >= 0) {
+      const otherCategory = newCategories.splice(otherCategoryIndex, 1)[0] as Category;
+      return { categories: newCategories, otherCategory };
+    }
+
+    return { categories: newCategories };
+  }
+}
+
+export interface Category {
```

Comment:
should we move this interfaces to its own .ts file?

Reply:
I don't know, maybe we can move them to separated files if we use them anywhere else. What do you think?

Reply:
As a former java user Im used to have classes/Interfaces splitted in different files but let @rubenmoya @ivanmalagon give their opinion